### PR TITLE
xy finance: Bridge : TVL by user

### PR DESCRIPTION
**Pull request by jaredWu0805**
## Title: 

- [x] Pattern for PR title "protocol_name : protocol_category : other comments". Example - "uniswap: dex : tvl by user"

## Checklist before requesting a review
1. **index.ts file**

     - [x] **Contains function**

            ```export const getUserTVLByBlock = async (blocks: BlockData) => {
                const { blockNumber, blockTimestamp } = blocks
                    //    Retrieve data using block number and timestamp
                    // YOUR LOGIC HERE
                
                return csvRows

            };
            ``` 
    - [x] **getUserTVLByBlock function takes input with this schema**
        
            ``` 
                interface BlockData {
                    blockNumber: number;
                    blockTimestamp: number;
                }
            ```
    - [x] **getUserTVLByBlock function returns output in this schema**

                ```
                const csvRows: OutputDataSchemaRow[] = [];

                type OutputDataSchemaRow = {
                    block_number: number;  //block_number which was given as input
                    timestamp: number;     // block timestamp which was given an input, epoch format
                    user_address: string;   // wallet address, all lowercase
                    token_address: string;  // token address all lowercase
                    token_balance: bigint;  // token balance, raw amount. Please dont divide by decimals
                    token_symbol: string; //token symbol should be empty string if it is not available
                    usd_price: number; //assign 0 if not available
                };
                ```
    - [x] **contains function**

            ```
                const readBlocksFromCSV = async (filePath: string): Promise<BlockData[]> => {
                const blocks: BlockData[] = [];

                await new Promise<void>((resolve, reject) => {
                    fs.createReadStream(filePath)
                    .pipe(csv()) // Specify the separator as '\t' for TSV files
                    .on('data', (row) => {
                        const blockNumber = parseInt(row.number, 10);
                        const blockTimestamp = parseInt(row.timestamp, 10);
                        if (!isNaN(blockNumber) && blockTimestamp) {
                        blocks.push({ blockNumber: blockNumber, blockTimestamp });
                        }
                    })
                    .on('end', () => {
                        resolve();
                    })
                    .on('error', (err) => {
                        reject(err);
                    });
                });

                return blocks;
                };

            ```
    - [x] **has this code**
           
            ```
            readBlocksFromCSV('hourly_blocks.csv').then(async (blocks: any[]) => {
            console.log(blocks);
            const allCsvRows: any[] = []; 

            for (const block of blocks) {
                try {
                    const result = await getUserTVLByBlock(block);
                    allCsvRows.push(...result);
                } catch (error) {
                    console.error(`An error occurred for block ${block}:`, error);
                }
            }
            await new Promise((resolve, reject) => {
                const ws = fs.createWriteStream(`outputData.csv`, { flags: 'w' });
                write(allCsvRows, { headers: true })
                    .pipe(ws)
                    .on("finish", () => {
                    console.log(`CSV file has been written.`);
                    resolve;
                    });
            });

            }).catch((err) => {
            console.error('Error reading CSV file:', err);
            });
        ```
2. **Output data**
    - [x] Data is returned for underlying tokens only. Not for special tokens (lp/veTokens etc)
    - [x] Follows the exact sequence mentioned in OutputDataSchemaRow . This is needed as we want same column ordering in output csv
    - Value of each field is :
        - [x] block_number *is same as input block number. This signifies TVL is as of this block_number.*
        - [x] timestamp is same as input timestamp. This signifies TVL is as of this timestamp. It is in epoch format.
        - [x] user_address is in lowercase
        - [x] token_address is in lowercase
        - [x] token_balance is in raw amount. Please dont divide by decimals.
        - [x] token_symbol value if present, empty string if value is not available.
        - [x] usd_price if value is available, 0 if value is not available.

### CODE REVIEW

**Comment by nitish-91** [comment]
@jaredWu0805  please create a branch from the main branch of this repo, make your changes in that branch and create pull request to main. 
https://github.com/delta-hq/l2-lxp-liquidity-reward#notes

**Comment by nitish-91** [comment]
> @jaredWu0805 please create a branch from the main branch of this repo, make your changes in that branch and create pull request to main. https://github.com/delta-hq/l2-lxp-liquidity-reward#notes

Please ignore this comment, we will figure this out internally

File: adapters/xyfinance/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,104 @@
+import { getEvents } from "./sdk/subgraphDetails";
+
+export const BATCH_SIZE = 1000;
+export const ETH_VAULT_ADDRESS = "0xa5cb30e5d30a9843b6481ffd8d8d35dded3a3251";
+export const ETH_VAULT_DEPLOY_BLOCK = 240;
+export const USDC_VAULT_ADDRESS = "0x9d90cfa17f3afcee2505b3e9d75113e6f5c9e843";
+export const USDC_VAULT_DEPLOY_BLOCK = 206280;
+export const VAULT_TO_TOKEN: Record<string, string> = {
```
Comment (0xroll) [inline adapters/xyfinance/src/index.ts:8]: @jaredWu0805  currently pools are hardcoded here. if there are new pools added, please submit a new pr to have them indexed

**Review by 0xroll — APPROVED** [review]

**Comment by 0xroll** [comment]
@RamonReis

**Comment by 0xroll** [comment]
@nitish-91 @RamonReis can we check if this is providing the correct output and merge this?
code wise looks ok

**Comment by nitish-91** [comment]
@jaredWu0805 we made one change in readme. Can you please go through it and incorporate. Sorry for the last minute addition.

Basically, we need this function to be defined and called in index.ts . We call this function to get the data by providing the hourly_blocks.csv file. Till now our engg was doing it, now it is stable and protocols adding it will reduce manual intervention from our side

[l2-lxp-liquidity-reward/adapters/example/dex/src/index.ts](https://github.com/delta-hq/l2-lxp-liquidity-reward/blob/607b15fb653d8d0780046e27014344880fce109d/adapters/example/dex/src/index.ts#L134)

**Comment by peterfangtw** [comment]
@nitish-91 a nudge on review 🙏

**Review by 0xroll — APPROVED** [review]

**Review by 0xroll — APPROVED** [review]

**Pull request merged**